components/indexer.py

```diff
@@ -1,61 +1,60 @@
 from magicbot import feedback
 import ctre
-import wpilib
 
 
 class Indexer:
     indexer_motors: list
-    indexer_switches: list
-    piston_switch: wpilib.DigitalInput
-    injector_switch: wpilib.DigitalInput
-    injector_master_motor: ctre.WPI_TalonSRX
-    injector_slave_motor: ctre.WPI_TalonSRX
 
     def setup(self):
         for motor in self.indexer_motors:
-            motor.setInverted(True)
+            motor.setInverted(False)
+            motor.setNeutralMode(ctre.NeutralMode.Brake)
+            motor.configForwardLimitSwitchSource(
+                ctre.LimitSwitchSource.FeedbackConnector,
+                ctre.LimitSwitchNormal.NormallyOpen,
+            )
 
-        self.indexer_speed = 0.2
-        self.injector_speed = 0.4
-
-        self.injector_slave_motor.follow(self.injector_master_motor)
-        self.injector_slave_motor.setInverted(ctre.InvertType.OpposeMaster)
+        self.indexer_speed = 0.6
 
     def on_enable(self) -> None:
-        self.indexing = True
+        self.intaking = True
 
     def execute(self) -> None:
-        if self.indexing:
-            # handle the injector motors
-            if self.injector_switch.get() and not self.piston_switch.get():
-                self.injector_master_motor.set(self.injector_speed)
-            else:
-                self.injector_master_motor.stopMotor()
-
-            # handle indexer motors
-            for motor, switch in zip(self.indexer_motors, self.indexer_switches):
-                if switch.get():
-                    motor.set(self.indexer_speed)
-                else:
-                    motor.stopMotor()
+        if self.intaking:
+            # Turn on all motors and let the limit switches stop it
+            for motor in self.indexer_motors:
+                motor.set(self.indexer_speed)
+
+            # Override any limit switches where the next cell is vacant
+            for first, second in zip(self.indexer_motors, self.indexer_motors[1:]):
+                first.overrideLimitSwitchesEnable(True)
+                at_limit = second.isFwdLimitSwitchClosed()
+                if not at_limit:
+                    first.overrideLimitSwitchesEnable(False)
```

Comment:
We probably want to only call `overrideLimitSwitchesEnable` once per motor controller in this loop.